Added erb view that lists custom emails

nrs1729 opened the pull request:
#### Describe the changes you have made in this PR:
I added a view to list the custom emails created so far (the view is only visible to admins). 

### Screenshots of the changes (If any):
![image](https://user-images.githubusercontent.com/51863168/70886523-906f5f80-1fd3-11ea-9056-1b1039bae902.png)
![image](https://user-images.githubusercontent.com/51863168/70980115-a6038880-20aa-11ea-8742-b57a6aeac720.png)

ankitkataria commented on config/routes.rb:
```diff
@@ -14,7 +14,7 @@
   end
 
-  resources :custom_mails, only: [:new, :create, :edit, :show, :update]
+  resources :custom_mails, only: [:index, :new, :create, :edit, :show, :update]
```
Can we fix this error?

ankitkataria commented on app/views/custom_mails/index.html.erb:
```diff
@@ -0,0 +1,30 @@
+            <% end %>
+            </tbody>
+          </table>
+        <% end %>
```
What if the mail count is 0? Could we have "No custom mails so far" is that case?

ankitkataria commented:
Also rebase this PR to latest master, only the commits relevant to the issue should be present in the branch.

nrs1729 commented:
Sorry about the commits, fixed it. I added a case for if there are no emails and a test that checks there's a response and that it contains the mail subject, is that okay? Not sure if there's something else to cover for the coverage/coveralls check. The string literals violations hound is bringing up seem in line with the rest of the code?

ankitkataria commented on spec/controllers/custom_mail_controller_spec.rb:
```diff
@@ -68,6 +68,14 @@
+
+    describe "#index" do
+      it "shows the list of custom views" do
+        get custom_mail_path(@mail)
```
You'll have to do `get custom_mails_path` to actually check the index route, this should check for the index route. 
This is also the reason why the test coverage is failing.

nrs1729 replied:
Thanks, should be fixed now
